packages/forms-web-app/__tests__/unit/controllers/full-appeal/submit-appeal/enter-code.test.js

```diff
@@ -35,7 +36,26 @@ describe('controllers/full-appeal/submit-appeal/enter-code', () => {
 			await getEnterCode(req, res);
 			expect(res.render).toBeCalledWith(`${ENTER_CODE}`, { requestNewCodeLink: url });
 		});
+
+		it('should redirect to appeal submitted page if appeal submitted', async () => {
+			req.session.appeal.state = 'SUBMITTED';
+			await getEnterCode(req, res);
+			expect(res.redirect).toBeCalledWith(`/${APPEAL_ALREADY_SUBMITTED}`);
+		});
+
+		it('should redirect to appeal submitted page if no appeal state', async () => {
+			req.session.appeal.state = null;
+			await getEnterCode(req, res);
+			expect(res.redirect).toBeCalledWith(`/${APPEAL_ALREADY_SUBMITTED}`);
+		});
+
+		it('should redirect to appeal submitted page if no appeal object', async () => {
```

Comment:
Why it is redirecting to "Appeal already submitted if no appeal was found?

Reply:
once a user submits an appeal, it removes the appeal object from req.session - therefore if we want the 'enter code' link in the email to redirect to the 'appeal already submitted' page once an appeal is submitted (given the 'enter-code' link doesn't contain any information about the specific appeal), we check if the appeal no longer exists. Same logic for the !appeal.state, because if the user has gone back to the before-you-start section, there will be a new appeal object in req.session but it won't have a state field yet. Basically, this is all a workaround for the fact the 'enter-code' link in the email is not actually linked to the specific appeal.